app/components/Poll/index.js

```diff
@@ -64,17 +64,39 @@ class Poll extends React.Component<Props, State> {
       : this.setState({ optionsToShow: options, expanded: true });
   };
 
+  optionsWithRatios = (options, optionsToShow) => {
+    const totalVotes = options.reduce((a, option) => a + option.votes, 0);
+    return optionsToShow.map(option => {
+      return {...option, "ratio": option.votes / totalVotes * 100}
+    });
+  };
+
+  // As described in: https://stackoverflow.com/questions/13483430/how-to-make-rounded-percentages-add-up-to-100
+  perfectRatios = (options) => {
+    const off = 100 - options.reduce((a, option) => a + Math.floor(option.ratio), 0);
+    return options
+        .sort((a, b) => {
```

Comment:
Since good old `Array.prototype.sort()` is in place, this can mutate the redux-state. You can use `lodash` `sortBy` (https://lodash.com/docs/4.17.11#sortBy), or just use `slice()` like this to take a shallow copy of the array:

```js
return options.slice().sort(...).map(...).sort(...)
```